enclave/tls_cert.c

```diff
@@ -64,13 +65,17 @@ static oe_result_t generate_x509_self_signed_certificate(
     config.ext_oid = (char*)oid;
     config.ext_oid_size = oid_size;
 
-    // allocate memory for cert output buffer
-    cert_buf = (uint8_t*)oe_malloc(OE_MAX_CERT_SIZE);
+    // allocate memory for cert output buffer and leave room for paddings
+    oe_cert_size =
```

Comment:
Do we know where does the max value come from? If there is a reasonable max value, should we add a check against the size of certificate then?

Reply:
I think the max value were from previous estimation. The generated certificates are usually 5K - 7K. @ryanhsu19 what is a typical `oe_cert_size` value? If it's less than 8K, we can bump up OE_MIN_CERT_SIZE. It has to cover version, serial number, signature algorithm, validity, signature, etc.

Reply:
Padding with 1/4 of (oid_size + public_key_buf_size + remote_report_buf_size) means the padding a variable. For EEID, `remote_report_buf_size` can be 16K which means padding 4K. That seems too big.

Reply:
IMO `remote_report_buf_size` is the most important factor, while others are near constant, so I reset `OE_MIN_CERT_SIZE` to `1024` and set `oe_cert_size` to `OE_MIN_CERT_SIZE + remote_report_buf_size*5/4`.

Both test cases in the `attestation_cert_api` test have a size of 5.1K and 5.5K, respectively, and the buffer size in my proposed version is around 6.8K in both cases.

A 4K padding for a 16K data is indeed large...let me try some other values.

Reply:
After considering both `remote_report_buf_size` and `public_key_buf_size` as major dynamic factors and doing different experiments with the value of `OE_MIN_CERT_SIZE` being 512, 1024 and 2048, I have decided to set `OE_MIN_CERT_SIZE` to 2048. Since there is enough room for padding, I have removed the weight on `remote_report_buf_size`.